test/distributed/org/apache/cassandra/distributed/test/AutoSnapshotTtlTest.java

```diff
@@ -0,0 +1,150 @@
+        properties.set(CassandraRelevantProperties.SNAPSHOT_MIN_ALLOWED_TTL_SECONDS, FIVE_SECONDS);
+    }
+
+    @AfterClass
+    public static void after()
+    {
+        properties.close();
+    }
+
+    /**
+     * Check that when auto_snapshot_ttl=5s, snapshots created from TRUNCATE are expired after 10s
+     */
+    @Test
+    public void testAutoSnapshotTTlOnTruncate() throws IOException, InterruptedException
+    {
+        try (Cluster cluster = build().withNodes(1)
+                                      .withConfig(c -> c.with(Feature.GOSSIP)
+                                                        .set("auto_snapshot_ttl", String.format("%ds", FIVE_SECONDS)))
+                                      .start())
+        {
+            IInvokableInstance instance = cluster.get(1);
+
+            cluster.schemaChange("CREATE KEYSPACE IF NOT EXISTS default WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};");
+            cluster.schemaChange("CREATE TABLE default.tbl (key int, value text, PRIMARY KEY (key))");
+
+            populate(cluster);
+
+            // Truncate Table
+            cluster.schemaChange("TRUNCATE default.tbl;");
+
+            // Check snapshot is listed after table is truncated
+            instance.nodetoolResult("listsnapshots").asserts().success().stdoutContains(ColumnFamilyStore.SNAPSHOT_TRUNCATE_PREFIX);
+
+            // Check snapshot is removed after 10s
+            Thread.sleep(2 * FIVE_SECONDS * 1000L);
+            cluster.get(1).nodetoolResult("listsnapshots").asserts().success().stdoutNotContains(ColumnFamilyStore.SNAPSHOT_TRUNCATE_PREFIX);
+        }
+    }
+
+    /**
+     * Check that when auto_snapshot_ttl=5s, snapshots created from TRUNCATE are expired after 10s
+     */
+    @Test
+    public void testAutoSnapshotTTlOnDrop() throws IOException, InterruptedException
+    {
+        try (Cluster cluster = build().withNodes(1)
+                                      .withConfig(c -> c.with(Feature.GOSSIP)
+                                                                  .set("auto_snapshot_ttl", String.format("%ds", FIVE_SECONDS)))
+                                      .start())
+        {
+            IInvokableInstance instance = cluster.get(1);
+
+            cluster.schemaChange("CREATE KEYSPACE IF NOT EXISTS default WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};");
+            cluster.schemaChange("CREATE TABLE default.tbl (key int, value text, PRIMARY KEY (key))");
+
+            populate(cluster);
+
+            // Drop Table
+            cluster.schemaChange("DROP TABLE default.tbl;");
+
+            // Check snapshot is listed after table is dropped
+            instance.nodetoolResult("listsnapshots").asserts().success().stdoutContains(ColumnFamilyStore.SNAPSHOT_DROP_PREFIX);
+
+            // Check snapshot is removed after 10s
+            Thread.sleep(2 * FIVE_SECONDS * 1000L);
```

Comment:
This should be replaced by `Awaitility`. Await.poll.until(...) with timeout more than 10s.

Reply:
Updated on both `testAutoSnapshotTTlOnTruncate` and `testAutoSnapshotTTlOnDrop`.

Did not update on `testAutoSnapshotTtlDisabled` because it needs to wait the full 10 seconds to ensure the auto snapshots are *not* removed.